File: Problems/Leetcode/3001-4000/3001-3100/3079.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int sumOfEncryptedInt(vector<int>& nums) {
        int ans = 0;
        for (int& num : nums) {
            int maxV = 0, cnt = 0;
            while (num) {
                maxV = max(maxV, num % 10);
                num /= 10;
                cnt++;
            }
            ans += stoi(string(cnt, char(maxV + '0')));
        }
        return ans;
    }
};

int main() {
    Solution sol;

    vector<int> vec {10, 21, 31};
    cout << sol.sumOfEncryptedInt(vec);

    return 0;
}

File: Problems/Leetcode/3001-4000/3001-3100/3079.py
from typing import List


class Solution:
    def sumOfEncryptedInt(self, nums: List[int]) -> int:
        ans = 0
        for num in nums:
            cnt = 0
            maxV = 0
            while num:
                maxV = max(maxV, num % 10)
                num //= 10
                cnt += 1

            ans += int(str(maxV) * cnt)
        return ans
